test/jdk/javax/swing/JButton/4659800/EnterKeyActivatesButton.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.event.KeyEvent;
+import java.awt.Robot;
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+
+/*
+ * @test
+ * @key headful
+ * @requires (os.family == "windows")
+ * @bug 4659800
+ * @summary Check whether typing <Enter> key generates
+ * ActionEvent on focused Button or not. This is applicable only for
+ * WindowsLookAndFeel and WindowsClassicLookAndFeel.
+ * @run main EnterKeyActivatesButton
+ */
+public class EnterKeyActivatesButton {
+    private volatile boolean buttonPressed;
+    private JFrame frame;
+
+    public static void main(String[] s) throws Exception {
+        EnterKeyActivatesButton test = new EnterKeyActivatesButton();
+        test.runTest();
+    }
+
+    private static void setLookAndFeel(String lafName) {
+        try {
+            UIManager.setLookAndFeel(lafName);
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Ignoring Unsupported L&F: " + lafName);
+        } catch (ClassNotFoundException | InstantiationException
+                | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void disposeFrame() {
+        if (frame != null) {
+            frame.dispose();
+            frame = null;
+        }
+    }
+
+    private void createUI() {
+        frame = new JFrame();
+        JPanel panel = new JPanel();
+        panel.add(new JTextField("Text field"));
+        JButton focusedButton = new JButton("Button1");
+        focusedButton.addActionListener(e -> buttonPressed = true);
+        panel.add(focusedButton);
+        panel.add(new JButton("Button2"));
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.add(panel);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+        focusedButton.requestFocusInWindow();
+    }
+
+    public void runTest() throws Exception {
+        Robot robot = new Robot();
+        robot.setAutoDelay(100);
+        for (UIManager.LookAndFeelInfo laf : UIManager.getInstalledLookAndFeels()) {
```

Comment:
I can suggest iterating over the filtered list of lafs:
```java
LookAndFeelInfo[] winlafs = Arrays.stream(lafs)
                                  .filter(e -> e.getName().startsWith("Windows"))
                                  .toArray(LookAndFeelInfo[]::new)
```
Or you can collect the result as list via `.collect(Collectors.toList())`.

Alternatively, you can use the `forEach` method with the body of the `for`-loop.